token/program-2022/src/error.rs

```diff
@@ -418,6 +433,49 @@ impl PrintProgramError for TokenError {
             TokenError::CiphertextArithmeticFailed => {
                 msg!("Ciphertext arithmetic failed")
             }
+            TokenError::PedersenCommitmentMismatch => {
+                msg!("Pedersen commitments did not match")
+            }
+            TokenError::RangeProofLengthMismatch => {
+                msg!("Range proof lengths did not match")
+            }
+            TokenError::IllegalBitLength => {
+                msg!("Illegal transfer amount bit length")
+            }
+            TokenError::FeeCalculation => {
+                msg!("Transfer fee calculation failed")
+            }
+        }
+    }
+}
+
+#[cfg(not(target_os = "solana"))]
+impl From<TokenProofGenerationError> for TokenError {
+    fn from(e: TokenProofGenerationError) -> Self {
+        match e {
+            TokenProofGenerationError::ProofGeneration(_) => TokenError::ProofGeneration,
+            TokenProofGenerationError::NotEnoughFunds => TokenError::InsufficientFunds,
+            TokenProofGenerationError::IllegalAmountBitLength => TokenError::IllegalBitLength,
+            TokenProofGenerationError::FeeCalculation => TokenError::FeeCalculation,
+        }
+    }
+}
```

Comment:
For the errors, this is probably fine. You could do ` ProofGeneration(#[from] TokenProofGenerationError)`, like in https://github.com/anza-xyz/agave/blob/ecb44d7bd7bcfcbf1f342eb7eaf2728e7b6eefc5/rpc-client-api/src/client_error.rs#L14, but that might end up being more confusing in the end

Reply:
Yeah makes sense. `ProofGeneration(#[from] TokenProofGenerationError)` seems more elegant to me, but this seems to need some restructuring of the existing `TokenError` unfortunately. It could also end up being more confusing as you suggested.